packages/flutter_tools/test/commands.shard/hermetic/http_host_availability_validator_test.dart

```diff
@@ -0,0 +1,82 @@
+// @dart = 2.9
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/doctor.dart';
+import 'package:flutter_tools/src/http_host_availability_validator.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+
+import '../../src/common.dart';
+import '../../src/fake_http_client.dart';
+
+void main() {
+  group('http host availability validator', () {
+    const List<String> operatingSystemsToTest = <String>['windows', 'macos', 'linux'];
+
+    test('all http hosts are available', () async {
+      /// A mock HTTP client that returns a 200 Successful response for
+      /// every request
+      final FakeHttpClient mockClient = FakeHttpClient.any();
+
+      // Run the check for all operating systems one by one
+      for(final String operatingSystem in operatingSystemsToTest) {
+        final HttpHostAvailabilityValidator httpHostValidator = HttpHostAvailabilityValidator(
+          platform: FakePlatform(operatingSystem: operatingSystem),
+          httpClient: mockClient,
+        );
+
+        // Run the validation check and get the results
+        final ValidationResult result = await httpHostValidator.validate();
+
+        // Check for only one information message
+        expect(result.messages..removeWhere(
```

Comment:
Here and elsewhere, i would try to make the checks a little less smart. You should start by listing out the expected messages exactly.

Reply:
Thanks. I changed the tests to check the type of the result [`ValidationType.installed`, `ValidationType.partial` or `ValidationType.notAvailable`].